### PR TITLE
Allow ignoring fields

### LINKED ISSUE #27: Add option to skip fields when creating the parser
Either as a flag to the metadata, or as an argument to for example skip all fields that end with a certain postfix.

### PR DESCRIPTION
**Pull request by antonsteenvoorden**
Fixes #27

### CODE REVIEW

File: tests/test_parse.py [hunk 1]
```diff
@@ -234,3 +236,34 @@ class TConfig:
 
         with raises(NotImplementedError):
             parse(TConfig, ["--a", "1", "1", "2"])
+
+
+class TestIgnore:
+    def test_ignore_default(self):
```
Comment (rogiervandergeer) [inline tests/test_parse.py]: Should we verify it breaks when an ignored field has no default?

File: src/pydargs/__init__.py [hunk 2]
```diff
@@ -43,6 +43,9 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None) -> Dataclass:
 def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
     parser = ArgumentParser()
     for field in fields(tp):
+        if field.metadata.get("ignore", False):
```
Comment (rogiervandergeer) [inline src/pydargs/__init__.py]: Given that the `metadata` dictionary is not purely for use with `pydargs`, but can be used by any other package / functionality, can we find a better name for this option that is clearly related to "parsing"? Something like "no_parsing", "skip_parsing", "no_arg"?

File: tests/test_parse.py [hunk 3]
```diff
@@ -234,3 +234,31 @@ class TConfig:
...
+        b: str = field(default="something", metadata={"ignore": True})
+        c: bool = field(default=False, metadata=dict(ignore=False, as_flags=True))
+        z: str = "dummy-for-late-binding-closure"
+
+    def test_ignore_default(self):
+        config = parse(self.Config, [])
+        assert config.a == 5
+        assert config.b == "something"
+        assert config.c == 5
+        assert config.z == "dummy-for-late-binding-closure"
```
Comment (rogiervandergeer) [inline tests/test_parse.py]:
```suggestion
        z: str = "dummy"

    def test_ignore_default(self):
        config = parse(self.Config, [])
        assert config.a == 5
        assert config.b == "something"
        assert config.c == 5
        assert config.z == "dummy"
```

No need to be too specific :D

File: src/pydargs/__init__.py (hunk 2 above)
Comment (antonsteenvoorden) [reply src/pydargs/__init__.py]: 👍

File: tests/test_parse.py (hunk 1 above)
Comment (antonsteenvoorden) [reply tests/test_parse.py]: Yes good one!

Comment (antonsteenvoorden) [reply tests/test_parse.py]: Wait no I thought about this before, but this was for those cheeky properties we set at a later time. For example if you are using setters and getters and want to have something prefixed with `_` 
e.g.
```python
@dataclass
class Config:
    _hidden_property: int = field(metadata=dict(ignore_arg=True))

@property
def hidden_property():
    return self._hidden_property

@hidden_property.setter
def hidden_property(value):
    self._hidden_property = value

```

then we don't want this to fail when nothing is provided when the config is instantiated, right?

Comment (rogiervandergeer) [reply tests/test_parse.py]: In the example above, using `parse` would always fail as `_hidden_property` isn't set when calling `__init__`.

File: README.md [hunk 4]
```diff
@@ -54,3 +54,15 @@ The base types are supported: `int`, `float`, `str`, `bool`, as well as:
...
+
+### Ignoring fields
+Fields can be ignored by adding the `ignore_arg` metadata field:
+```python
+@dataclass
+class Config:
+    number: int
+    ignored: str = field(metadata=dict(ignore_arg=True))
+```
+When indicated, this field is not added to the parser and cannot be overridden via the cli.
```
Comment (rogiervandergeer) [inline README.md]:
```suggestion
When indicated, this field is not added to the parser and cannot be overridden with an argument.
```

**Review by rogiervandergeer — APPROVED** [review]

**Pull request merged**